src/components/Main/Main.tsx

```diff
@@ -0,0 +1,69 @@
+import { useState } from "react";
+
+import { FilmList } from "../FilmList";
+import { FilterMoviesList } from "../FilterMoviesList";
+import { genreList } from "../../films";
+import { Film, Modal } from "../../interfaces";
+import { CONSTANTS } from "../../constants";
+
+import "./style.scss";
+import { Form } from "../Form";
+import { filterByGenre, sorting } from "../../helpers";
+import { ErrorBoundary } from "../ErrorBoundary";
+
+interface MainProps {
+  modalState: Modal;
+  closeModal(): void;
+  movies: Film[];
+  openModal(type: string, filmID: string): void;
+}
+
+const Main: React.FC<MainProps> = ({
+  modalState,
+  closeModal,
+  movies,
+  openModal,
+}) => {
+  const [selectedGenre, setGenre] = useState(
+    CONSTANTS.DEFAULT_FILTERS.defaultGenre
+  );
+  const [sortBy, setSort] = useState("");
+
+  const onChangeGenre = (event: React.MouseEvent) => {
+    setSort("");
+    setGenre((prevGenre) =>
+      (event.target as HTMLInputElement).id
+        ? (event.target as HTMLInputElement).id
+        : prevGenre
```

Comment:
setGenre((prevGenre) => event.target.id || prevGenre))
try to increase 'genre' pixel area, if you sometime don't receive id field